### PR TITLE
CRM-19092 fix issue where  can be '' rather than 0 so cid is not set …

### PR DESCRIPTION
**Pull request by seamuslee001**
…properly

### CODE REVIEW

**Comment by seamuslee001** [comment]
To reproduce the error, 
1. Login as an administer user that has the ability to record contributions on behalf of others. 
2, Navigate to an online contribution form to record a membership
3, Click the Not me link
4. Leave the on behalf of other box empty and fill out the form with different details to the contact record of the logged in user
5. Complete the transaction and find submitted data overwrote the logged in user's contact record. 

I found that $tempID some how was getting set to be '' not sure exactly how but this fix attempts to capture it.

**Comment by seamuslee001** [comment]
this looks like to be identical issue as found in https://issues.civicrm.org/jira/browse/CRM-17844

**Comment by seamuslee001** [comment]
@eileenmcnaughton this is just backporting code in 4.7, It fixes the issue it seems on 4.6 for me on AUG test box. I will await confirmation from Andrew but i think this is safe to backport

**Comment by seamuslee001** [comment]
@eileenmcnaughton Just noting I have released this patch to AUG prod

File: CRM/Core/Form.php [hunk 1]
```diff
@@ -1584,7 +1584,7 @@ public static function validateMandatoryFields($fields, $values, &$errors) {
    */
   public function getContactID() {
     $tempID = CRM_Utils_Request::retrieve('cid', 'Positive', $this);
-    if (isset($this->_params) && isset($this->_params['select_contact_id'])) {
+    if (isset($this->_params) && !empty($this->_params['select_contact_id'])) {
```
Comment (liedekef) [inline CRM/Core/Form.php]: Would this not better be (don't assume select_contact_id is set):
 if (isset($this->_params, $this->_params['select_contact_id']) && !empty($this->_params['select_contact_id'])) {

**Comment by liedekef** [comment]
I added a line note on this change, it seems to me this might trigger a php notice

File: CRM/Core/Form.php (hunk 1 above)
Comment (seamuslee001) [inline CRM/Core/Form.php]: @liedekef Franky i think that is something we can look at however this is the exact replica of the code that is in 4.7 aka master, I haven't yet spotted it generating a notice on our dev box but i agree that it feels strange (thanks for the code review tho)

**Comment by seamuslee001** [comment]
@liedekef Franky i did some research, it appears according to here http://php.net/manual/en/function.empty.php that empty essentially includes an isset test as well. But as the documentation shows it won't throw a php notice if what its testing is not set.

**Comment by liedekef** [comment]
You're correct, I just tested this locally and indeed empty doesn't throw a php notice if not set (go figure ... apparently the case since php 5.5). So ignore my comment :-)

**Comment by eileenmcnaughton** [comment]
OK merging based on the discussion thread appearing to agree on this and it being a backport of upstream code.

**Pull request merged**